### PR TITLE
hplip: hardcode ppdc path to fix #44230 and add -n to gzip to improve reproducibility

### LINKED ISSUE #44230: hplipWithPlugin: ppdc not found
## Issue description
I expect all of these to not return "No such file or directory" and I expect the "bad interpreter" to be patched.

I think it still "works", but compilation output should be clean such that when debugging something in the future there is not first a mess of other warnings to go through.

Multiple issues:

```
ppdc: Unable to find include file "<font.defs>" on line 261 of prnt/drv/hpcups.drv.
gzip: ppd/hpcups/*.ppd: No such file or directory
ppdc: Unable to find include file "<font.defs>" on line 60 of prnt/drv/hpijs.drv.
gzip: ppd/hpijs/*.ppd: No such file or directory
./createPPD.sh: ./hpijs-drv: /usr/bin/perl: bad interpreter: No such file or directory
./configure: line 16070: test: too many arguments
```

### Steps to reproduce
```
  services.printing = {
      enable = true;
      drivers = [ pkgs.hplipWithPlugin ];
  };
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.55, NixOS, 18.03.git.6115f47 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

### PR DESCRIPTION
**Pull request by Artturin**
the ppdc: Warning - overlapping filename can be ignored
https://bugs.launchpad.net/hplip/+bug/1756967
```
ppdc: Warning - overlapping filename "ppd/hpijs/hp-color_laserjet_3500-hpijs.ppd".
ppdc: Warning - overlapping filename "ppd/hpijs/hp-color_laserjet_3500n-hpijs.ppd".
ppdc: Warning - overlapping filename "ppd/hpijs/hp-color_laserjet_3550-hpijs.ppd".
ppdc: Warning - overlapping filename "ppd/hpijs/hp-color_laserjet_3550n-hpijs.ppd".
ppdc: Warning - overlapping filename "ppd/hpijs/hp-color_laserjet_3600-hpijs.ppd".
```

###### Motivation for this change
Closes #44230

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128812` at 0e087952 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>hplip</li>
    <li>hplipWithPlugin</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `coreutils`.
Near pkgs/misc/drivers/hplip/default.nix:6:9:
```
  |
6 | , bash, coreutils, util-linux
  |         ^
```
</li>
  </ul>
</details>

File: pkgs/misc/drivers/hplip/default.nix [hunk 1]
```diff
@@ -102,6 +102,11 @@ python3Packages.buildPythonApplication {
   ];
 
   prePatch = ''
+    # issue 44230
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/hplip/default.nix:105]: Please link the full issue.

File: pkgs/misc/drivers/hplip/default.nix [hunk 2]
```diff
@@ -149,6 +154,12 @@ python3Packages.buildPythonApplication {
     export CUPS_DATADIR="${cups}/share/cups"
   '';
 
+  postConfigure = ''
+    # don't save timestamp, in order to improve reproducibility
+    substituteInPlace ./Makefile \
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/hplip/default.nix:159]:
```suggestion
    substituteInPlace Makefile \
```

File: pkgs/misc/drivers/hplip/default.nix [hunk 3]
```diff
@@ -102,6 +102,11 @@ python3Packages.buildPythonApplication {
   ];
 
   prePatch = ''
+    # issue 44230
+    substituteInPlace ./createPPD.sh \
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/hplip/default.nix:106]:
```suggestion
    substituteInPlace createPPD.sh \
```

**Comment by Artturin** [comment]
well that was a mess

File: pkgs/misc/drivers/hplip/default.nix [hunk 4]
```diff
@@ -102,6 +102,11 @@ python3Packages.buildPythonApplication {
   ];
 
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/hplip/default.nix:104]:
```suggestion
  postPatch = ''
```

**Pull request merged**